File: 05_Arrays/src/Transpose_Matrix_867.js
const transpose = function (matrix) {
  let ans = [];
  for (let i = 0; i < matrix[0].length; i++) {
    let temp = new Array(matrix.length).fill(0);
    ans.push(temp);
  }

  for (let i = 0; i < ans.length; i++) {
    for (let j = 0; j < ans[i].length; i++) {
      ans[i][j] = matrix[j][i];
    }
  }

  return ans;
};

console.log(transpose([1, 2, 3], [4, 5, 6], [7, 8, 9]));

File: 05_Arrays/src/Transpose_Matrix_867.java
import java.util.Arrays;

public class Transpose_Matrix_867 {
    public static void main(String[] args) {
        int[][] ans = transpose(new int[][] {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}});
        for (int[] arr : ans) {
            System.out.println(Arrays.toString(arr));
        }
    }
    public static int[][] transpose(int[][] matrix) {
        int[][] ans = new int[matrix[0].length][matrix.length];

        for (int i = 0; i < ans.length; i++) {
            for (int j = 0; j < ans[i].length; j++) {
                ans[i][j] = matrix[j][i];
            }
        }

        return ans;
    }
}
